core/src/test/java/org/apache/calcite/test/SqlToRelConverterTest.java

```diff
@@ -4690,6 +4690,23 @@ void checkUserDefinedOrderByOver(NullCollation nullCollation) {
         .ok();
   }
 
+  /** Test case for:
+   * <a href="https://issues.apache.org/jira/browse/CALCITE-6013">[CALCITE-6013]
+   * Unnecessary measures added as projects during rel construction</a>.
+   */
+  @Test void testAvoidUnnecessaryMeasureProject() {
+    final String sql = "select deptno\n"
+        + "from empm\n"
+        + "group by deptno";
```

Comment:
The `GROUP BY` is needed here to trigger the bug correct? Otherwise `createAggImpl` is not called?

Are we certain this issue only impacted aggregate queries?

Reply:
Yeah that is my understanding. As a quick test I removed the `GROUP BY` and put the offending lines back in `SqlToRelConverter`. No error.